### PR TITLE
Add survey, question, answer types to survey resolvers

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened
Add survey, question, answer types to survey resolvers
 
## Insight
- Add all fields for each type (survey, question, answer)
- Create separated resolver for getting survey details
- Support [pagination](https://graphql-ruby.org/pagination/using_connections.html) for survey list
- Add tests for resolvers

## Proof Of Work
GraphiQL

### CODE REVIEW

File: app/graphql/resolvers/survey_resolver.rb [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
-    description 'Survey list'
+    description 'Survey details'
+    argument :id, ID, required: true
+    type Types::SurveyType, null: true
 
-    type [Types::SurveyType], null: false
-
-    def resolve
-      Survey.all
+    def resolve(id:)
```
Comment (team-nimblehq) [inline app/graphql/resolvers/survey_resolver.rb:9]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/634d923a312cee5a5351ccd312a92b5bc125132a/app/graphql/resolvers/survey_resolver.rb#L9"></span>Doesn’t depend on instance state (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/graphql/resolvers/surveys_resolver.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Resolvers
+  class SurveysResolver < BaseResolver
+    description 'Survey list'
+    type Types::SurveyType.connection_type, null: false
+
+    def resolve
```
Comment (team-nimblehq) [inline app/graphql/resolvers/surveys_resolver.rb:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/634d923a312cee5a5351ccd312a92b5bc125132a/app/graphql/resolvers/surveys_resolver.rb#L8"></span>Doesn’t depend on instance state (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/graphql/types/base/connection.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Types
+  module Base
+    class Connection < GraphQL::Types::Relay::BaseConnection
+      field :totalCount, Integer, null: false
+
+      def totalCount
```
Comment (team-nimblehq) [inline app/graphql/types/base/connection.rb:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/e16fb7196db4de5a797840ca93bdb6f34b32271b/app/graphql/types/base/connection.rb#L8"></span>Has the name ‘totalcount’</td>
    </tr>
  </tbody>
</table>

File: app/graphql/resolvers/surveys_resolver.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Resolvers
+  class SurveysResolver < BaseResolver
+    description 'Survey list'
+    type Types::SurveyType.connection_type, null: false
```
Comment (olivierobert) [inline app/graphql/resolvers/surveys_resolver.rb:6]: What does this do? I am also curious of `connection_type_class Types::Base::Connection` in a file below. Why is this needed? Thanks for enlightening me in advance 🙇

Comment (olivierobert) [reply app/graphql/resolvers/surveys_resolver.rb:6]: Nevermind I saw the link to https://graphql-ruby.org/pagination/using_connections.html

File: app/graphql/types/survey_type.rb [hunk 5]
```diff
@@ -9,9 +9,10 @@ class SurveyType < Types::Base::Object
...
     field :is_active, Boolean, null: true
     field :cover_image_url, String, null: true
-    field :created_at, String, null: true
-    field :active_at, String, null: true
-    field :inactive_at, String, null: true
-    field :type, String, null: true
+    field :created_at, GraphQL::Types::ISO8601DateTime, null: true
+    field :active_at, GraphQL::Types::ISO8601DateTime, null: true
+    field :inactive_at, GraphQL::Types::ISO8601DateTime, null: true
+    field :survey_type, String, null: true, method: :type
```
Comment (olivierobert) [inline app/graphql/types/survey_type.rb:15]: Is this the same field `type` as before? What is `method: :type` for? IS it for the pagination? I am confused here 😵

**Review by olivierobert — APPROVED** [review]
